sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1312,9 +1312,15 @@ RTDeviceBinaryImage *getBinImageFromMultiMap(
         reinterpret_cast<const char *>(&CompileTargetByteArray[0]),
         CompileTargetByteArray.size());
     // Note: there are no explicit targets for CPUs, so on x86_64,
-    // so we use a spir64_x86_64 compile target image.
+    // intel_cpu_spr, and intel_cpu_gnr, we use a spir64_x86_64
+    // compile target image.
+    // TODO: When dedicated targets for CPU are added, (i.e.
+    // -fsycl-targets=intel_cpu_spr etc.) remove this special
+    // handling of CPU targets.
     if ((ArchName == CompileTarget) ||
-        (ArchName == "x86_64" && CompileTarget == "spir64_x86_64")) {
+        ((ArchName == "x86_64" || ArchName == "intel_cpu_spr" ||
+          ArchName == "intel_cpu_gnr") &&
+         CompileTarget == "spir64_x86_64")) {
```

Comment:
Ideally, we should swap `ArchName` and `CompileTarget` checks so that we perform less string comparisons. For example in case where we have several AOT-compiled device images for GPU and we are trying to launch some kernel on CPU.